odoo/addons/base/models/ir_http.py

```diff
@@ -210,7 +210,7 @@ def _pre_dispatch(cls, rule, args):
                 args[key].check_access_rule('read')
             except (odoo.exceptions.AccessError, odoo.exceptions.MissingError) as e:
                 # custom behavior in case a record is not accessible / has been removed
-                if handle_error := rule.endpoint.original_routing.get('handle_params_access_error'):
+                if handle_error := rule.endpoint.original_routing.get('handle_params_access_error') or rule.endpoint.routing.get('handle_params_access_error'):
```

Comment:
In my case the value for `handle_params_access_error` will come from `rule.endpoint.routing.get('handle_params_access_error')` which mean the original method not from my extend method, so i don't think we need that. 
Also check at python cmd too
![image](https://github.com/odoo/odoo/assets/56789189/d5eac122-b1fa-45f2-8f6a-678327e6b937)

Reply:
Ah I think this slipped from my original review (of PR #135926), we shouldn't access `original_routing` because that's the "function's specific `@route`". `routing` is the one holding the "merged `@route` accros inherited controllers". I think the following will just work:

```diff
diff --git a/odoo/addons/base/models/ir_http.py b/odoo/addons/base/models/ir_http.py
index d63ec1400a64..058965d152bc 100644
--- a/odoo/addons/base/models/ir_http.py
+++ b/odoo/addons/base/models/ir_http.py
@@ -210,7 +210,7 @@ class IrHttp(models.AbstractModel):
                 args[key].check_access_rule('read')
             except (odoo.exceptions.AccessError, odoo.exceptions.MissingError) as e:
                 # custom behavior in case a record is not accessible / has been removed
-                if handle_error := rule.endpoint.original_routing.get('handle_params_access_error'):
+                if handle_error := rule.endpoint.routing.get('handle_params_access_error'):
                     if response := handle_error(e):
                         werkzeug.exceptions.abort(response)
                 if isinstance(e, odoo.exceptions.MissingError):
```

Reply:
Yes, it does work and make more sense. shorten the code. Thanks